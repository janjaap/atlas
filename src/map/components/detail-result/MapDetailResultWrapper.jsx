import React from 'react';
import PropTypes from 'prop-types';

const MapDetailResultWrapper = ({
  children, panoUrl, subTitle, title, onMaximize, onPanoPreviewClick
}) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <button
          className={`
            map-detail-result__header-pano-button
            map-detail-result__header-pano-button--${panoUrl ? 'enabled' : 'disabled'}
          `}
          disabled={!panoUrl}
          title={panoUrl ? 'Panoramabeeld tonen' : 'Geen Panoramabeeld beschikbaar'}
          onClick={onPanoPreviewClick}
        >
          <img
            alt="Panoramabeeld"
            className="map-detail-result__header-pano"
            height="292"
            src={panoUrl}
            width="438"
          />
        </button>
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">{title}</h1>
        {subTitle && (
          <h2 className="map-detail-result__header-subtitle">{subTitle}</h2>
        )}
      </div>
    </header>
    <div className="map-detail-result__scroll-wrapper">
      {children && (
          [children]
      )}
      <footer className="map-search-results__footer">
        <button
          className="map-search-results__button"
          onClick={onMaximize}
          title="Volledig weergeven"
        >
          <span className="
            map-search-results__button-icon
            map-search-results__button-icon--maximize"
          />
          Volledig weergeven
        </button>
      </footer>
    </div>
  </section>
);

MapDetailResultWrapper.defaultProps = {
  children: null,
  subTitle: '',
  onPanoPreviewClick: /* istanbul ignore next */ () => null
};

MapDetailResultWrapper.propTypes = {
  children: PropTypes.element,
  panoUrl: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func
};

export default MapDetailResultWrapper;
