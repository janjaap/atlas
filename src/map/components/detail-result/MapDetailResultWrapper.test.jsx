import React from 'react';
import { shallow } from 'enzyme';

import MapDetailResultWrapper
  from './MapDetailResultWrapper';

describe('MapDetailResultWrapper', () => {
  it('should render everything', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailResultWrapper
        panoUrl="panoUrl"
        title="title"
        subTitle="subTitle"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
      >
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render empty panoUrl', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailResultWrapper
        panoUrl=""
        title="title"
        subTitle="subTitle"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
      >
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render empty sub title', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailResultWrapper
        panoUrl=""
        title="title"
        subTitle=""
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
      >
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render missing sub title', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailResultWrapper
        panoUrl=""
        title="title"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
      >
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render missing children', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailResultWrapper
        panoUrl=""
        title="title"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
