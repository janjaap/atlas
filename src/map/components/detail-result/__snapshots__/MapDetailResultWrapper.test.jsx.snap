// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailResultWrapper should not render empty panoUrl 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        title
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        subTitle
      </h2>
    </div>
  </header>
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <span
          className=" map-search-results__button-icon map-search-results__button-icon--maximize"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render empty sub title 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        title
      </h1>
    </div>
  </header>
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <span
          className=" map-search-results__button-icon map-search-results__button-icon--maximize"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render missing children 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        title
      </h1>
    </div>
  </header>
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <span
          className=" map-search-results__button-icon map-search-results__button-icon--maximize"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render missing sub title 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        title
      </h1>
    </div>
  </header>
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <span
          className=" map-search-results__button-icon map-search-results__button-icon--maximize"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should render everything 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--pano
      "
  >
    <button
      className="
            map-detail-result__header-pano-button
            map-detail-result__header-pano-button--enabled
          "
      disabled={false}
      onClick={[MockFunction]}
      title="Panoramabeeld tonen"
    >
      <img
        alt="Panoramabeeld"
        className="map-detail-result__header-pano"
        height="292"
        src="panoUrl"
        width="438"
      />
    </button>
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        title
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        subTitle
      </h2>
    </div>
  </header>
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <span
          className=" map-search-results__button-icon map-search-results__button-icon--maximize"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;
